app/controllers/api/messaging.php

```diff
@@ -3251,19 +3251,68 @@
             throw new Exception(Exception::MESSAGE_NOT_FOUND);
         }
 
+        if ($status !== MessageStatus::DRAFT && \count($topics) === 0 && \count($users) === 0 && \count($targets) === 0) {
+            throw new Exception(Exception::MESSAGE_MISSING_TARGET);
+        }
+
+        $currentScheduledAt = $message->getAttribute('scheduledAt');
+
         switch ($message->getAttribute('status')) {
             case MessageStatus::PROCESSING:
                 throw new Exception(Exception::MESSAGE_ALREADY_PROCESSING);
             case MessageStatus::SENT:
                 throw new Exception(Exception::MESSAGE_ALREADY_SENT);
             case MessageStatus::FAILED:
                 throw new Exception(Exception::MESSAGE_ALREADY_FAILED);
+            case MessageStatus::SCHEDULED:
+                if (\is_null($scheduledAt) && \is_null($currentScheduledAt)) {
+                    throw new Exception(Exception::MESSAGE_MISSING_SCHEDULE);
+                }
+                break;
```

Comment:
What if the message was draft and then I update it to be scheduled, but I don't pass `$scheduledAt`?

Reply:
Should schedule even be a status??? I really doubt this.

Reply:
Resolved by https://github.com/appwrite/appwrite/pull/7670